lib/foreman/renderer.rb

```diff
@@ -57,6 +59,11 @@ def foreman_url(action = "built")
               :token     => (@host.token.value unless @host.try(:token).nil?), :kind => action
     end
 
+    def foreman_server_fqdn
+      config = URI.parse(Setting[:unattended_url])
+      config.host || request.host
```

Comment:
Templating port is irrelevant for Discovery as it always uses HTTPS port and HTTP port (which is utilized for templating). Therefore this won't solve the discovery case at all. Maybe if you could add a param `force_https` that could be used in Discovery global PXELinux section to force it.